torchao/quantization/quant_api.py

```diff
@@ -1592,24 +1588,28 @@ def _int8_dynamic_activation_int8_weight_quantize_tensor(weight, config):
             _layout=layout,
             zero_point_domain=weight_zero_point_domain,
         )
-        quantized_weight = to_linear_activation_quantized(
-            quantized_weight, input_quant_func
-        )
+        quantized_weight = to_linear_activation_quantized(new_weight, input_quant_func)
     else:
         from torchao.quantization.quantize_.workflows.int8.int8_tensor import (
             QuantizeTensorToInt8Kwargs,
         )
 
+        assert config.granularity in {PerRow(), PerTensor()}, (
+            "Only PerRow and PerTensor are supported"
+        )
+        weight_granularity, act_granularity = _normalize_granularity(config.granularity)
```

Comment:
`_normalize_granularity` includes both normalization and validation right now I think and it's float8 specific, I feel we should split normalize and validation and validation should be Int8 specific, see: https://github.com/pytorch/ao/blob/16aad7c2f4073d415a541cbfd00aa40fe4ef7162/torchao/prototype/float8_opaque_tensor/inference_workflow.py#L60-L62 for example